nabla-indexer/test/indexer_tests.ts

```diff
@@ -0,0 +1,1067 @@
+    if (condition(indexerRouters)) {
+      return indexerRouters;
+    }
+
+    await new Promise((resolve) => setTimeout(resolve, 500));
+  }
+  return undefined;
+}
+
+function assertFeeHistoriesEqual(actual: IndexerFeesHistory, expected: IndexerFeesHistory) {
+  assertEq(actual.backstopFees, expected.backstopFees);
+  assertEq(actual.lpFees, expected.lpFees);
+  assertEq(actual.protocolFees, expected.protocolFees);
+}
+
+export default async function (environment: TestSuiteEnvironment) {
+  const {
+    address,
+    unit,
+    tester,
+    vm,
+    constructors: { newRouter, newERC20Wrapper, newSwapPool, newBackstopPool, newMockOracle, newNablaCurve },
+  } = environment;
+
+  type NablaInstance = ReturnType<typeof createNablaInstance> extends Promise<infer T> ? T : never;
+
+  const createNablaInstance = async (initialBackstopPoolLiquidity: bigint, intialSwapPoolLiqudity: bigint[]) => {
+    const noOfSwapPools = intialSwapPoolLiqudity.length;
+
+    const nablaCurve = await newNablaCurve(0, e(0.01, 18));
+    const router = await newRouter();
+
+    const assets = [await newERC20Wrapper("Test US Dollar", "USD", 12, [1], [0], [], [])];
+    for (let i = 1; i <= noOfSwapPools; i++) {
+      assets.push(await newERC20Wrapper(`Token ${i}`, `Tok${i}`, 12, [1], [i], [], []));
+    }
+
+    for (let i = 0; i <= noOfSwapPools; i++) {
+      const mintExtrinsic = vm.extrinsicBuilders.tokens.setBalance(tester, { XCM: i }, unit(1000), 0);
+      await vm.executeRootExtrinsic(mintExtrinsic);
+    }
+
+    const oracles: TestContract[] = [];
+    for (let i = 0; i <= noOfSwapPools; i++) {
+      const oracle = await newMockOracle(address(assets[i]), unit(i + 1));
+      oracles.push(oracle);
+      await router.setPriceOracle(address(assets[i]), address(oracle));
+    }
+
+    const backstop = await newBackstopPool(address(router), address(assets[0]), "Backstop LP", "BLP");
+
+    const TREASURY = "6k9LbZKC3dYDqaF6qhS9j438Vg1nawD98i8VuHRKxXSvf1rp";
+    const swapPools: TestContract[] = [];
+    for (let i = 1; i <= noOfSwapPools; i++) {
+      const swapPool = await newSwapPool(
+        address(assets[i]),
+        address(nablaCurve),
+        address(router),
+        address(backstop),
+        TREASURY,
+        `LP ${i}`,
+        `LP${i}`
+      );
+      swapPools.push(swapPool);
+
+      await router.registerPool(address(assets[i]), address(swapPool));
+      await backstop.addSwapPool(address(swapPool), 0);
+    }
+
+    await assets[0].approve(address(backstop), MAX_UINT256);
+    for (let i = 1; i <= noOfSwapPools; i++) {
+      await assets[i].approve(address(router), MAX_UINT256);
+      await assets[i].approve(address(swapPools[i - 1]), MAX_UINT256);
+    }
+
+    const BOB = "6k6gXPB9idebCxqSJuqpjPaqfYLQbdLHhvsANH8Dg8GQN3tT";
+    vm.startPrank(BOB);
+    await assets[0].approve(address(backstop), MAX_UINT256);
+    for (let i = 1; i <= noOfSwapPools; i++) {
+      await assets[i].approve(address(router), MAX_UINT256);
```

Comment:
Is it also required for the depositor to give approval for the router contract?

Reply:
Indeed that should not be required for a deposit. It is only required for a token swap. I think I wanted to do this here preemptively as I possibly planned to use it in one of the tests (i.e., that BOB would execute a swap). I didn't check whether it's still required. Should not harm to leave it in.

Reply:
Ah makes sense. No it does not hurt at all just wanted to be sure I understood.